### PR TITLE
feat!: allow disabling of seqsets [CFG]

### LINKED ISSUE #3210: Allow disabling of SeqSets
As a part of #2104, make it possible to disable SeqSets:
* (most important) hide the SeqSets navigation item on the website
* disable SeqSet pages on the website
* disable the `SeqSetCitationsController` in the backend
* make it configurable in the Helm value.yaml

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #3210

preview URL: https://3210-allow-disabling-of-s.loculus.org/

### Summary

I grouped the seqSet config values. The values.yaml section can look like
```
seqSets:
  enabled: false
``` 
or
```
seqSets:
  enabled: false
  crossRef: null
``` 
or
```
seqSets:
  enabled: true
  crossRef:
    DOIPrefix: "placeholder"
    endpoint: "https://test.crossref.org"
``` 

When `seqSets. enabled == false`, 
* the backend endpoint are disabled
* the website SeqSets navigation item is disabled
* the SeqSets pages on the website return 404

BREAKING CONFIG CHANGES:
* `enableCrossRefCredentials` was removed. Use `seqSets.enabled` instead.
* moved `crossRef` to `seqSets.crossRef`

### Screenshot

grafik

grafik

### PR Checklist

- [x] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: kubernetes/loculus/values.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 environment: server
 robotsNoindexHeader: false
-crossRef:
-  DOIPrefix: "placeholder"
-  endpoint: "https://test.crossref.org"
+seqSets:
+  enabled: false
```
Comment (fengelniederhammer) [inline kubernetes/loculus/values.yaml:4]: Reminder: revert this before merge

**Comment by theosanderson** [comment]
(We eventually coalesced on `feat!` for breaking changes and have some CI that uses that)

File: kubernetes/loculus/values.yaml [hunk 2]
```diff
@@ -1519,7 +1521,6 @@ secrets:
     data:
       username: "dummy"
       password: "dummy"
-enableCrossRefCredentials: true
```
Comment (theosanderson) [inline kubernetes/loculus/values.yaml:1522]: (Using this line to create a thread on a general theme)

There are valid use cases for SeqSets without the ability to generate DOIs (i.e. without CrossRef). I'm sure those weren't supported in a super nice way before this (e.g. the DOI button was always shown even if it didn't work) but it seems like this PR probably makes that harder? I think we should maintain the possibility to run SeqSets without  specifying CrossRef credentials (if nothing else, for testing) [that may already be maintained, I'm not clear].

**Comment by theosanderson** [comment]
> Is that actually documented somehwere?

👍 https://github.com/loculus-project/loculus/pull/3213

File: kubernetes/loculus/values.yaml (hunk 2 above)
Comment (fengelniederhammer) [reply kubernetes/loculus/values.yaml:1522]: Indeed, I didn't have that in mind. Should I change it such that one can have SeqSets without CrossRef?

Comment (theosanderson) [reply kubernetes/loculus/values.yaml:1522]: IMO we shouldn't change it to make it _harder_ to do that unless we have a clear reason to. I don't see one atm. I don't think we have a reason atm to invest in making SeqSets work better without crossref than they do atm (e.g. disabling the generate DOI button) but we could do.

Comment (theosanderson) [reply kubernetes/loculus/values.yaml:1522]: (we can probably still remove the `enableCrossRef credentials` thing and make it directly look for the specification of the secret in the yaml, which I think I did for something else)

Comment (fengelniederhammer) [reply kubernetes/loculus/values.yaml:1522]: Now it's possible to set `seqSets.crossRef = null`: Then you have SeqSets, but no CrossRef integration. Locally, it worked for me.

**Comment by fengelniederhammer** [comment]
There is still something weird with the config... On the preview, I can't enable the CrossRef integration. I need to check tomorrow. Locally everything seems to work as expected.

**Comment by fengelniederhammer** [comment]
Sorry for the noise - I tried to make the CrossRef configuration a bit better (it should warn you when it's not properly configured). But Spring acted weird, I don't know why I couldn't get it working in Kubernetes. I removed those changes how.

File: .idea/runConfigurations/BackendApplicationKt.run.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
     <module name="backend.main" />
-    <option name="PROGRAM_PARAMETERS" value="--spring.datasource.url=jdbc:postgresql://localhost:5432/loculus --spring.datasource.username=postgres --spring.datasource.password=unsecure --keycloak.user=backend --keycloak.password=backend --keycloak.realm=loculus --keycloak.client=backend-client --keycloak.url=http://localhost:8083 --loculus.config.path=website/tests/config/backend_config.json --loculus.debug-mode=true --spring.security.oauth2.resourceserver.jwt.jwk-set-uri=http://localhost:8083/realms/loculus/protocol/openid-connect/certs" />
+    <option name="PROGRAM_PARAMETERS" value="--spring.datasource.url=jdbc:postgresql://localhost:5432/loculus --spring.datasource.username=postgres --spring.datasource.password=unsecure --keycloak.user=backend --keycloak.password=backend --keycloak.realm=loculus --keycloak.client=backend-client --keycloak.url=http://localhost:8083 --loculus.config.path=website/tests/config/backend_config.json --loculus.debug-mode=true --spring.security.oauth2.resourceserver.jwt.jwk-set-uri=http://localhost:8083/realms/loculus/protocol/openid-connect/certs --loculus.enable-seqset-citations=false" />
```
Comment (theosanderson) [inline .idea/runConfigurations/BackendApplicationKt.run.xml:4]: Q: will this disable seqsets for dev by default? If so I think that's not ideal

File: kubernetes/loculus/templates/loculus-backend.yaml [hunk 4]
```diff
@@ -41,6 +41,8 @@ spec:
           ports:
             - containerPort: 8079
           args:
+            - "--loculus.enable-seqset-citations={{ $.Values.seqSets.enabled }}"
```
Comment (theosanderson) [inline kubernetes/loculus/templates/loculus-backend.yaml:44]: (not blocking) I find this name "enable-seqset-citations" weird - its seqsets themselves that are affected not merely "seqset citations" (which I would think of as about citations of seqsets, tracking of which is a feature yet to come

File: kubernetes/loculus/values.yaml (hunk 2 above)
Comment (theosanderson) [reply kubernetes/loculus/values.yaml:1523]: Thanks!

File: website/src/components/Navigation/SandwichMenu.tsx
Comment (theosanderson) [inline website/src/components/Navigation/SandwichMenu.tsx:1]: thanks for the refactor

File: .idea/runConfigurations/BackendApplicationKt.run.xml (hunk 3 above)
Comment (fengelniederhammer) [reply .idea/runConfigurations/BackendApplicationKt.run.xml:4]: Indeed. I changed it to enabled

File: kubernetes/loculus/templates/loculus-backend.yaml (hunk 4 above)
Comment (fengelniederhammer) [reply kubernetes/loculus/templates/loculus-backend.yaml:44]: I'll rename to `enable-seqsets'

**Pull request merged**